### ISSUE TITLE
runtime: clean up async preemption loose ends

### ISSUE THREAD

**Issue by aclements** [golang/go#36365]
Go 1.14 introduces asynchronous preemption so that tight loops can be preempted (#10958). However, there are still several loose ends to tie up. This is a general bug to keep track of remaining work:

- [ ] Support on all possible platforms (js/wasm is not currently possible)
  - [ ] windows/arm ([CL 207961](https://golang.org/cl/207961))
  - [ ] darwin/arm
  - [ ] plan9/*
- [ ] Redo unsafe-point representation so we can eliminate register maps
- [ ] Remove register maps and extra stack maps (after redesigning debug call injection)
- [ ] Possibly incorporate @cherrymui's sequence restarting ([CL 208126](https://golang.org/cl/208126))
- [ ] Make large pointer-free memmoves/memclrs preemptible (#31222)
- [ ] Make large pointer-full memmoves/memclrs preemptible (this is a little harder)
- [ ] Fix various annoying spins in the scheduler
- [ ] Make more of the runtime preemptible

**Comment by ianlancetaylor** [comment]
Is this still a 1.15 issue?

**Comment by aclements** [comment]
Moving to 1.16 for the remaining tasks.

**Comment by networkimprov** [comment]
Has a `go build` switch to disable async preemption been considered?

Some folks are having trouble with it, and can't rely on the env variable to disable it... #40722

**Comment by ianlancetaylor** [comment]
@cherrymui Anything to be done here for 1.17?  Thanks.

**Comment by cherrymui** [comment]
No, I don't think we did or plan to do anything on this for 1.17.

**Comment by bcmills** [comment]
@golang/runtime, does the checklist on this issue need to be updated? I see that https://go.dev/cl/366735 and https://go.dev/cl/366734 were merged in 2021 (for `windows/arm64` and `windows/arm`).
